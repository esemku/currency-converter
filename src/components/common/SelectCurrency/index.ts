export { default as SelectCurrency } from './SelectCurrency';
