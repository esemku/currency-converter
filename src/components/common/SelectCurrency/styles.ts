import { makeStyles } from '@material-ui/core';
import { colors, fonts } from 'theme';

export default makeStyles((theme) => ({
  root: {
    position: 'relative',
    marginBottom: 16,
    width: '100%',
    '& .MuiInputBase-input': {
      padding: 0,
    },
    '& .MuiInputBase-root, .MuiInputBase-root::before': {
      border: 'none',
      borderBottom: 'none !important',
    },
    '& .MuiInput-underline::after': {
      border: 'none',
    },
  },
  select: {
    backgroundColor: colors.grayTertiary,
    color: colors.grayQuaternary,
    transition: 'color .2s ease, background-color .2s ease',
    border: `1px solid ${colors.grayTertiary}`,
    borderRadius: theme.shape.borderRadius,
    width: '100%',
    padding: '10px 18px',
    fontSize: fonts.size.mobile.input,
    fontWeight: fonts.weight.mobile.input,
    lineHeight: fonts.lineHeight.mobile.input,
    [theme.breakpoints.up('md')]: {
      fontSize: fonts.size.desktop.input,
      fontWeight: fonts.weight.desktop.input,
      lineHeight: fonts.lineHeight.desktop.input,
    },
    marginTop: 6,
    marginBottom: 6,
    '&:focus': {
      backgroundColor: colors.blackSecondary,
    },
  },
  selectMenu: {
    '& .MuiMenu-paper': {
      backgroundColor: colors.graySecondary,
    },
    '& .MuiListItem-root.Mui-selected': {
      backgroundColor: colors.bluePrimary,
      color: colors.whitePrimary,
    },
  },
  selectMenuItem: {
    color: colors.grayQuaternary,
    fontSize: 13,
    fontWeight: 400,
    lineHeight: '19.5px',
    minHeight: 'unset',
    padding: '10px 16px',
    '&:hover': {
      backgroundColor: colors.blueQuaternary,
      color: colors.bluePrimary,
    },
  },
  label: {
    color: colors.whitePrimary,
    fontSize: fonts.size.mobile.inputLabel,
    fontWeight: fonts.weight.mobile.inputLabel,
    lineHeight: fonts.lineHeight.mobile.inputLabel,
    [theme.breakpoints.up('md')]: {
      fontSize: fonts.size.desktop.inputLabel,
      fontWeight: fonts.weight.desktop.inputLabel,
      lineHeight: fonts.lineHeight.desktop.inputLabel,
    },
  },
  currencyFlag: {
    marginRight: 16,
  },
  currencyName: {
    opacity: 0.6,
  },
}));
