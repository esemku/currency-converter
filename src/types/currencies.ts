export interface ICurrency {
  fromCurrency: string;
  toCurrency: string;
  amount: number;
}

export interface ICurrencyListItem {
  iso: string;
  currency_name: string;
  is_obsolete: boolean;
}
