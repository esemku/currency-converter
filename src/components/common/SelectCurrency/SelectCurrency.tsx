import { MenuItem, Select } from '@material-ui/core';
import { nanoid } from 'nanoid';
import React from 'react';
import CurrencyFlag from 'react-currency-flags';
import { ICurrencyListItem } from 'types/currencies';
import { IconChevronDown } from '../icons';
import useStyles from './styles';

interface IProps {
  currenciesList: ICurrencyListItem[];
  value: string;
  label: string;
  id: string;
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
}

const SelectCurrency: React.FC<IProps> = ({
  currenciesList,
  value,
  label,
  onChange,
  id,
}) => {
  const styles = useStyles();

  return (
    <div className={styles.root}>
      <label htmlFor={id} className={styles.label}>
        {label}
      </label>
      <Select
        id={id}
        value={value}
        onChange={onChange}
        IconComponent={IconChevronDown}
        className={styles.select}
        MenuProps={{ className: styles.selectMenu }}
        data-testid={`selectCurrency${id}`}
      >
        {currenciesList.map((currencyItem: ICurrencyListItem) => (
          <MenuItem
            value={currencyItem.iso}
            key={nanoid()}
            className={styles.selectMenuItem}
          >
            <CurrencyFlag
              currency={currencyItem.iso}
              size="sm"
              className={styles.currencyFlag}
            />
            {currencyItem.iso}
            <span className={styles.currencyName}>
              &nbsp;-&nbsp;{currencyItem.currency_name}
            </span>
          </MenuItem>
        ))}
      </Select>
    </div>
  );
};

export default SelectCurrency;
