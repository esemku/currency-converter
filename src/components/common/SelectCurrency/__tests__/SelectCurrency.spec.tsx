import { render } from '@testing-library/react';
import React from 'react';
import { ICurrencyListItem } from 'types/currencies';
import SelectCurrency from '../SelectCurrency';

const currenciesList: ICurrencyListItem[] = [
  { iso: 'PLN', currency_name: 'Pol', is_obsolete: false },
  { iso: 'GBP', currency_name: 'Eng', is_obsolete: false },
  { iso: 'NOK', currency_name: 'Nor', is_obsolete: false },
];

describe('<SelectCurrency />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SelectCurrency
        currenciesList={currenciesList}
        value="PLN"
        label="label"
        id="id"
        onChange={jest.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
