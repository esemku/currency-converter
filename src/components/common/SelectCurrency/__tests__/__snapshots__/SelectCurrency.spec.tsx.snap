// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectCurrency /> renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <label
      class="makeStyles-label-5"
      for="id"
    >
      label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-2"
      data-testid="selectCurrencyid"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="id"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="id"
        role="button"
        tabindex="0"
      >
        <span
          class="sc-bdfBQB sc-gsTEea jczEZi fzESjn makeStyles-currencyFlag-6"
        />
        PLN
        <span
          class="makeStyles-currencyName-7"
        >
           - 
          Pol
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="PLN"
      />
      <svg
        class="MuiSelect-icon"
        fill="none"
        height="20"
        style="margin-top: 1px;"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.434 12.734L7.25 8.55a1.06 1.06 0 00-1.5 1.5l5.543 5.543a1 1 0 001.414 0l5.543-5.543a1.06 1.06 0 10-1.5-1.5l-4.184 4.184a.8.8 0 01-1.132 0z"
          fill="#565674"
        />
      </svg>
    </div>
  </div>
</div>
`;
